prometheuspvesd/discovery.py

```diff
@@ -85,19 +85,26 @@ def _get_variables(self, pve_list, pve_type):
 
         return variables
 
-    def _get_ip_address(self, pve_type, pve_node, vmid):
+    def _get_ip_addresses(self, pve_type, pve_node, vmid):
 
-        def validate(address):
+        def validate_ipv4(address: object) -> object:
             try:
-                # IP address validation
-                if socket.inet_aton(address):
-                    # Ignore localhost
-                    if address != "127.0.0.1":
-                        return address
-            except socket.error:
+                if ipaddress.ip_address(address) not in ipaddress.ip_network("127.0.0.0/8"):
+                    return address
+            except ValueError:
                 return False
 
-        address = False
+        def validate_ipv6(address: object) -> object:
+            try:
+                # Make sure to skip loopback addresses and local-link addresses
+                if ipaddress.ip_address(address) not in ipaddress.ip_network("::1/128") \
```

Comment:
There is also `is_loopback`. Or the other way around, we could just check `is_private` and `is_global` for valid private or public addresses?

Reply:
I think using `is_loopback` and `is_link_local` is the proper way to go, since `is_private` would only allow usage of `10.0.0.0/8`, `172.16.0.0/12` and `192.168.0.0/16` since ipv6 doesn't really have private networks. For `is_global` it's the other way around, and these three networks would be dropped, which probably are the most common ones for vms to run in